src/dsp_tools/utils/xmlupload/stash_circular_references.py

```diff
@@ -18,122 +24,92 @@
 logger = get_logger(__name__)
 
 
-def _stash_circular_references(
-    nok_resources: list[XMLResource],
-    ok_res_ids: set[str],
-) -> tuple[list[XMLResource], set[str], list[XMLResource], Stash | None]:
+def stash_circular_references(
+    resources: list[XMLResource],
+    stash_lookup: dict[str, list[str]],
+) -> Stash | None:
     """
-    Raises:
-        BaseError
+    Stashes problematic resource-references from a list of resources.
+    The resources are modified in-place.
+
+    Args:
+        resources: all resources of the XML file
+        stash_lookup: A dictionary which maps the resources that have stashes to the UUIDs of the stashed links
+
+    Returns:
+        stash: an object that contains the stashed references
     """
     stashed_standoff_values: list[StandoffStashItem] = []
     stashed_link_values: list[LinkValueStashItem] = []
-    ok_resources: list[XMLResource] = []
 
-    for res in nok_resources.copy():
+    for res in resources.copy():
+        if not res.id in stash_lookup:
+            continue
         for link_prop in res.get_props_with_links():
             if link_prop.valtype == "text":
```

Comment:
how about factoring out everything in the `if x == "text"` and `if x == resptr` into a helper function, so that this function gets a bit more readable?